src/ui/public/vislib/lib/dispatch.js

```diff
@@ -186,28 +185,32 @@ export default function DispatchClass(Private, config) {
     addBrushEvent(svg) {
       if (!this.isBrushable()) return;
 
+      const self = this;
       const xScale = this.handler.xAxis.xScale;
       const brush = this.createBrush(xScale, svg);
 
-      function brushEnd() {
+      function simulateClickWithBrushEnabled(d, i) {
         if (!validBrushClick(d3.event)) return;
-
-        const bar = d3.select(this);
-        const startX = d3.mouse(svg.node());
-        const startXInv = xScale.invert(startX[0]);
-
-        // Reset the brush value
-        brush.extent([startXInv, startXInv]);
-
-        // Magic!
-        // Need to call brush on svg to see brush when brushing
-        // while on top of bars.
-        // Need to call brush on bar to allow the click event to be registered
-        svg.call(brush);
-        bar.call(brush);
+        if (xScale.invert) {
+          const bar = d3.select(this);
+          const startX = d3.mouse(svg.node());
+          const startXInv = xScale.invert(startX[0]);
+
+          // Reset the brush value
+          brush.extent([startXInv, startXInv]);
+
+          // Magic!
+          // Need to call brush on svg to see brush when brushing
+          // while on top of bars.
+          // Need to call brush on bar to allow the click event to be registered
+          svg.call(brush);
+          bar.call(brush);
+        } else {
+          self.emit('click', self.eventResponse(d, i));
```

Comment:
Can you add a comment explaining this condition? E.g. what does the lack of the `xScale.invert` property imply and how must we respond to it?